puma/hlplots/aux_results.py

```diff
@@ -120,192 +151,188 @@ def add_taggers_from_file(  # pylint: disable=R0913
             # add tagger to results
             self.add(tagger)
 
-    def plot_var_vtx_eff(
+    def __getitem__(self, tagger_name: str):
+        """Retrieve Tagger object.
+
+        Parameters
+        ----------
+        tagger_name : str
+            Name of model
+
+        Returns
+        -------
+        Tagger
+            Instance of the puma.hlplots.Tagger class, containing tagger information.
+        """
+        return self.taggers[tagger_name]
+
+    def get_filename(self, plot_name: str, suffix: str | None = None):
+        """Get output name.
+
+        Parameters
+        ----------
+        plot_name : str
+            plot name
+        suffix : str, optional
+            suffix to add to output name, by default None
+
+        Returns
+        -------
+        str
+            output name
+        """
+        base = f"{self.sample}_{plot_name}"
+        if suffix is not None:
+            base += f"_{suffix}"
+        return Path(self.output_dir / base).with_suffix(f".{self.extension}")
+
+    def plot_var_vtx_perf(
```

Comment:
I moved the previous set of plotting functions into a single function that is meant to produce all relevant vertexing plots for a single sample/flavour. Probably will need to converge on which plots make sense to produce by default in future.